crates/core/tedge/src/cli/init.rs

```diff
@@ -1,26 +1,102 @@
 use crate::command::BuildContext;
 use crate::command::Command;
+use crate::Component;
+use anyhow::bail;
 use anyhow::Context;
+use clap::Subcommand;
+use std::os::unix::fs::MetadataExt;
+use std::path::Path;
 use tedge_utils::file::create_directory;
 use tedge_utils::file::PermissionEntry;
 
 #[derive(Debug)]
 pub struct TEdgeInitCmd {
     user: String,
     group: String,
+    relative_links: bool,
     context: BuildContext,
 }
 
 impl TEdgeInitCmd {
-    pub fn new(user: String, group: String, context: BuildContext) -> Self {
+    pub fn new(user: String, group: String, relative_links: bool, context: BuildContext) -> Self {
         Self {
             user,
             group,
+            relative_links,
             context,
         }
     }
 
     fn initialize_tedge(&self) -> anyhow::Result<()> {
+        let executable_name =
+            std::env::current_exe().context("retrieving the current executable name")?;
+        let stat = std::fs::metadata(&executable_name).with_context(|| {
+            format!(
+                "reading metadata for the current executable ({})",
+                executable_name.display()
+            )
+        })?;
+        let Some(executable_dir) = executable_name.parent() else {
+            bail!(
+                "current executable ({}) does not have a parent directory",
+                executable_name.display()
+            )
+        };
+
+        let component_subcommands: Vec<String> =
+            Component::augment_subcommands(clap::Command::new("tedge"))
+                .get_subcommands()
+                .map(|c| c.get_name().to_owned())
+                .chain(["tedge-apt-plugin".to_owned()])
+                .collect();
+
+        for component in &component_subcommands {
+            let link = executable_dir.join(component);
+            match std::fs::symlink_metadata(&link) {
+                Err(e) if e.kind() != std::io::ErrorKind::NotFound => bail!(
+                    "couldn't read metadata for {}. do you need to run with sudo?",
+                    link.display()
+                ),
+                meta => {
+                    let file_exists = meta.is_ok();
+                    if file_exists {
+                        nix::unistd::unlink(&link).with_context(|| {
+                            format!("removing old version of {component} at {}", link.display())
+                        })?;
+                    }
+
+                    let tedge = if self.relative_links {
+                        // TODO this could lead to broken links
+                        // TODO is this feature even that useful?
+                        Path::new("./tedge")
```

Comment:
question: What are some use cases where using relative links could be useful?

Reply:
Honestly, I'm not entirely sure, hence the comment immediately above it. I guess it makes moving the binaries a tad easier, but the fact that `tedge init` sets the links up should solve that problem in any case. I guess if someone wants to store tedge on some non-deterministically named mount point or something that might be a possible use case, but I am leaning towards declaring this feature pointless and removing it to simplify the code and CLI.

Reply:
I've slightly refactored this, but I think I'll keep this, as it doesn't add much complexity, and there may be valid use cases. Absolute paths are probably the right default, and what `tedge-apt-plugin` uses (although for that case, a relative path wouldn't make any sense).